### PR TITLE
Predicate helpers

### PR DESCRIPTION
**Pull request by ababkin**

### CODE REVIEW

File: src/Network/Eureka/Version/Cabal.hs [hunk 1]
```diff
@@ -3,14 +3,20 @@ module Network.Eureka.Version.Cabal (
...
-import           Network.Eureka.Version.Types (Version)
-import           Text.ParserCombinators.ReadP (readP_to_S)
+import Data.Maybe (listToMaybe)
+import qualified Data.Version as DV (Version, Version (Version),
+  parseVersion, showVersion)
+import Network.Eureka.Version.Types (Version)
+import Text.ParserCombinators.ReadP (readP_to_S)
+import Network.Eureka.Version.Types (Predicate)
 
+versionInRangeP :: [Int] -> [Int] -> Predicate
```
Comment (MHova) [inline src/Network/Eureka/Version/Cabal.hs]: Have you considered restricting the types here to `(Int, Int, Int, Int)`?

Comment (ababkin) [inline src/Network/Eureka/Version/Cabal.hs]: That would prescribe that there must always be 4 versions, which may not always be desirable.
We can of course introduce our own constraints, but before over-constraining ourselves at the outset, I decided to go with same constraints used in https://hackage.haskell.org/package/base-4.8.0.0/docs/Data-Version.html that is the underlying version lib i'm currently using for the eureka-semver lib.

Comment (MHova) [inline src/Network/Eureka/Version/Cabal.hs]: Okay. Is it an error case if the two lists provided are of different lengths? If so, should we fail fast and explode?

Comment (ababkin) [inline src/Network/Eureka/Version/Cabal.hs]: i dont think it should be an error. I think it should work the same way as in cabal. e.g. 1.1.2 < 1.2

Comment (MHova) [inline src/Network/Eureka/Version/Cabal.hs]: :+1:

**Comment by ababkin** [comment]
@MHova redone the formatting

**Comment by MHova** [comment]
Thanks :+1:

**Pull request merged**